handlers/handler.go

```diff
@@ -101,9 +101,28 @@ func (env *Env) PostUserHandler(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(reqUser)
 }
 
+// PatchUserHandler updates specific columns of given user
+func (env *Env) DeleteUserHandler(w http.ResponseWriter, r *http.Request) {
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil || userID <= 0 {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	err = env.DB.DeleteUser(userID)
+	if err != nil {
+		errMsg := "Failed to delete user"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusNotFound)
+		internalServerError(w, err)
```

Comment:
```suggestion
		internalServerError(w, err)
		return
```
You can just respond with an Internal Server Error and then `return`.